### ISSUE TITLE
FMU export "author", "license" and "copyright" data

### ISSUE THREAD

**Issue by BasilioPV** [OpenModelica/OpenModelica#13169]
### Description

The "author", "license" and "copyright" data of a FMU is not exported by OpenModelica:

![Screenshot_2024-11-10-21-52-33-016_cn.wps.moffice_eng.xiaomi.lite.png](https://github.com/user-attachments/assets/f21dd8ab-9d01-462c-8fbe-8329ddba9ca3)

In addition, there are no specific annotations in Modelica language that covers these data (except for library encryption).

### Expected Behavior

I suggest to create custom vendor annotations to define the three missing values and use them in the FMU export as follows:

__OpenModelica_FMUauthor = "some author"
__OpenModelica_FMUlicense = "some license"
__OpenModelica_FMUcopyright = "some copyright"

**Comment by casella** [comment]
@BasilioPV how would you like to set these annotations in the GUI?

**Comment by BasilioPV** [comment]
@casella I suggest to include as custom vendor annotations on the model to be exported to FMU:

![image](https://github.com/user-attachments/assets/7aa1497e-0daa-4234-bba1-f0dcb7d88b14)

In the same way used on the documentation export, OM does not use any extends to find this info on other models used.

**Comment by arun3688** [comment]
@BasilioPV I can implement this @casella are you ok with the above annotation name suggestions ?

**Comment by casella** [comment]
One important point before starting the implementation. In your example, you apply all this information to a single _model_. This is actually wrong for the uses and version annotation, that must be set on a top-level entity, usualy an enclosing package, see [MLS Section 18.8.2](https://specification.modelica.org/maint/3.6/annotations.html#version-handling)

The question is, what should we do about this with author, licence and copyright information. I'm not sure you want to add this information to each and every model in a package. Or maybe you need that, specifying different authors for models belonging to the same package. This can actually be useful information, besides FMI. One option could be that if such an annotation is not found in the class being turned into an FMU, then OMEdit looks in the enclosing classes, all the way up to the top package.

Also, I'm not totally sure we need to add "FMU" to those custom annotation names. Maybe `__OpenModelica_author` etc. would be enough?

**Comment by BasilioPV** [comment]
@casella The main idea here is to set these atributes to the FMU when exported. I only will add these attributes to those models that I want to export to FMU (not to all models of a package).

Regarding to version, a single model can be a top-level class, as shown in the MSL example:

<img width="288" alt="image" src="https://github.com/user-attachments/assets/10b0ab80-b545-4eea-91ba-2c802ec46bd2">


Respect to this: "One option could be that if such an annotation is not found in the class being turned into an FMU, then OMEdit looks in the enclosing classes, all the way up to the top package.", I suggest to not search hierarchically upwards, because this can produce unwanted results.

For me, removing FMU is also correct:
__OpenModelica_author = "some author"
__OpenModelica_license = "some license"
__OpenModelica_copyright = "some copyright"

From my point of view, these attributes are applicable for the derived work of the single model itself, and not to the inner components used (that shall be mentioned in the license attribute).

**Comment by casella** [comment]
> @casella The main idea here is to set these atributes to the FMU when exported. I only will add these attributes to those models that I want to export to FMU (not to all models of a package).

OK.
 
> Regarding to version, a single model can be a top-level class

Sure. However, people normally collect models within packages. That's what I always recommend to my students. Loose collections of top-level models quickly become messy.

> Respect to this: "One option could be that if such an annotation is not found in the class being turned into an FMU, then OMEdit looks in the enclosing classes, all the way up to the top package.", I suggest to not search hierarchically upwards, because this can produce unwanted results.

OK.

> For me, removing FMU is also correct: __OpenModelica_author = "some author" __OpenModelica_license = "some license" __OpenModelica_copyright = "some copyright"
> 
> From my point of view, these attributes are applicable for the derived work of the single model itself, and not to the inner components used (that shall be mentioned in the license attribute).

To me, this is obvious. The copyright of the models that you _use_ remains with their respective copyright holders. Whether or not you are authorized to use them in your model is another story.

**Comment by arun3688** [comment]
@BasilioPV the issue is fixed with this commit 496dd7381cda25e236a4ba33451fcaf941a39c08, Please test it from the nightly builds tomorrow (14-11-2024)

**Comment by BasilioPV** [comment]
@arun3688 Tested and work as expected. Good job!
